cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -45,7 +45,7 @@
 /**
  * Base class for program runtime context
  */
-public abstract class AbstractContext extends AbstractServiceDiscoverer implements DatasetContext, RuntimeContext {
+public abstract class AbstractContext extends AbstractStreamWriter implements DatasetContext, RuntimeContext {
```

Comment:
This doesn't fix the problem I'd commented earlier since it exposes the stream writing capabilities to all program contexts (which, as per demo, we don't want to allow). 

This class should continue extending `AbstractServiceDiscoverer` and *only* `BasicWorkerContext` should extend `AbstractStreamWriter`.